src/main/java/org/openmaptiles/layers/Water.java

```diff
@@ -86,21 +105,56 @@ record WaterInfo(int minZoom, int maxZoom, String clazz) {}
       case "ne_110m_ocean" -> new WaterInfo(0, 1, FieldValues.CLASS_OCEAN);
       case "ne_50m_ocean" -> new WaterInfo(2, 4, FieldValues.CLASS_OCEAN);
       case "ne_10m_ocean" -> new WaterInfo(5, 5, FieldValues.CLASS_OCEAN);
-
-      // TODO: get OSM ID from low-zoom natural earth lakes
-      case "ne_110m_lakes" -> new WaterInfo(0, 1, FieldValues.CLASS_LAKE);
-      case "ne_50m_lakes" -> new WaterInfo(2, 3, FieldValues.CLASS_LAKE);
-      case "ne_10m_lakes" -> new WaterInfo(4, 5, FieldValues.CLASS_LAKE);
       default -> null;
     };
     if (info != null) {
-      features.polygon(LAYER_NAME)
-        .setBufferPixels(BUFFER_SIZE)
-        .setZoomRange(info.minZoom, info.maxZoom)
-        .setAttr(Fields.CLASS, info.clazz);
+      setupNeWaterFeature(features, info.minZoom, info.maxZoom, info.clazz, null);
+      return;
+    }
+
+    LakeInfo lakeInfo = switch (table) {
+      case "ne_110m_lakes" -> new LakeInfo(0, 1, FieldValues.CLASS_LAKE);
+      case "ne_50m_lakes" -> new LakeInfo(2, 3, FieldValues.CLASS_LAKE);
+      case "ne_10m_lakes" -> new LakeInfo(4, 5, FieldValues.CLASS_LAKE);
+      default -> null;
+    };
+    if (lakeInfo != null) {
+      try {
+        var geom = feature.worldGeometry();
+        lakeInfo.geom = geom;
+        lakeInfo.name = feature.getString("name");
+
+        var neLakeNameMap = neLakeNameMaps.computeIfAbsent(table, t -> new ConcurrentHashMap<>());
+
+        // need to externally synchronize inserts into ArrayList
+        synchronized (this) {
+          neAllLakeInfos.add(lakeInfo);
+        }
+        neLakeIndex.put(geom, lakeInfo);
+        if (lakeInfo.name != null) {
+          if (!neLakeNameMap.containsKey(lakeInfo.name) ||
+            lakeInfo.geom.getArea() > neLakeNameMap.get(lakeInfo.name).geom.getArea()) {
+            // on name collision, bigger lake gets on the name list
+            neLakeNameMap.put(lakeInfo.name, lakeInfo);
+          }
```

Comment:
I think we want to do an atomic operation on the concurrent hashmap here to avoid multiple updates at the same time from colliding with eachother:

```java
neLakeNameMap.merge(lakeInfo.name, (prev, next) -> next.geom.getArea() > prev.geom.getArea() ? next : prev);
```